core/src/main/java/com/linecorp/armeria/client/WebClient.java

```diff
@@ -312,15 +352,50 @@ static WebClientBuilder builder(URI uri) {
         return new WebClientBuilder(uri);
     }
 
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@code protocol}
+     * and base {@link EndpointGroup}.
+     *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(String protocol, EndpointGroup endpointGroup) {
+        return builder(SessionProtocol.of(requireNonNull(protocol, "protocol")), endpointGroup);
+    }
+
     /**
      * Returns a new {@link WebClientBuilder} created with the specified {@link SessionProtocol}
      * and base {@link EndpointGroup}.
      *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(SessionProtocol protocol, EndpointGroup endpointGroup) {
+        return builder(protocol, endpointGroup, null);
+    }
+
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@code protocol}.
+     * base {@link EndpointGroup} and path.
+     *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(String protocol, EndpointGroup endpointGroup, @Nullable String path) {
```

Comment:
How about making `path` non-null? Because this builder already have 
`builder(SessionProtocol protocol, EndpointGroup endpointGroup)`.

Reply:
Made `path` non-null. :bowing_man: